### PR TITLE
[improve][txn]Add log when can't send TcClientConnectRequest to broker.

### PR DESCRIPTION
**Pull request by liangyepianzhou**
Main Issue: https://github.com/apache/pulsar/issues/20851
### Motivation
When the Proto version does not allow us to send TcClientConnectRequest to the broker, we should add a log to debug it.

### Modifications

Add a waining log.
### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by poorbarcode — APPROVED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java [hunk 1]
```diff
@@ -167,6 +167,8 @@ public CompletableFuture<Void> connectionOpened(ClientCnx cnx) {
                     return null;
                 });
             } else {
+                LOG.warn("Can not Connect to the transaction coordinator because the protocol version {} is "
```
Comment (poorbarcode) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java:170]: I saw the compare line is `if (cnx.getRemoteEndpointProtocolVersion() > ProtocolVersion.v18.getValue()) `, why log "lower than 19?"

Comment (Technoboy-) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java:170]: `Connect` -> `connect`

**Review by lifepuzzlefun — APPROVED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java [hunk 2]
```diff
@@ -167,6 +167,8 @@ public CompletableFuture<Void> connectionOpened(ClientCnx cnx) {
                     return null;
                 });
             } else {
+                LOG.warn("Can not connect to the transaction coordinator because the protocol version {} is "
+                                + "lower than 19", cnx.getRemoteEndpointProtocolVersion());
                 registerToConnection(cnx);
```
Comment (lifepuzzlefun) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java:172]: Just curious that if client protocol version lower than the expected version？does any txn request will throw exception？It seems the Future just completed as normal

Comment (liangyepianzhou) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java:172]: The broker version should keep consistent if the client proto version is lower than 18.

**Pull request merged**